lib/cmd/common/admin-devices.js

```diff
@@ -13,108 +12,97 @@ devices.usage = "\nfhc devices list"
+  if (args.length === 0) {
     return list(cb);
   }
 
-  var action = args[0];
-  if (action === 'list') {
-    return list(cb);
-  }else if (action === 'read'){
-    if (args.length !== 2) return cb("Invalid arguments for 'read':" + devices.usage);
-    return read(args[1], cb);
-  }else if (action === 'update'){
-    if (args.length !== 3) return cb("Invalid arguments for 'update':" + devices.usage);
-    return update(args[1], args[2], cb);
-  }else if (action === 'enable'){
-    if (args.length !== 3) return cb("Invalid arguments for 'enable':" + devices.usage);
-    return enabledDevice(args[1], args[2], cb);
-  }else if (action === 'purgedata'){
-    if (args.length !== 3) return cb("Invalid arguments for 'purgedata':" + devices.usage);
-    return purgedataDevice(args[1], args[2], cb);
-  }else if (action === 'listusers'){
-    if (args.length !== 2) return cb("Invalid arguments for 'listusers':" + devices.usage);
-    return listusers(args[1], cb);
-  }else if (action === 'listapps'){
-    if (args.length !== 2) return cb("Invalid arguments for 'listapps':" + devices.usage);
-    return listapps(args[1], cb);
-  }else if (args.length === 1){
-    var deviceId = args[0];
-    return read(deviceId, cb);
-  }else{
+  var action = args.shift();
+  var func = actions[action];
+  if (!func) {
+    // command not found, try `read({deviceId})`
+    if (args.length === 1) {
+      var deviceId = args[0];
+      return read(deviceId, cb);
+    }
     return cb(devices.usage);
   }
+  if (args.length !== func.length + 1) {
+    return cb("Invalid arguments for '" + action + "':" + devices.usage);
+  }
+  return func.command.apply(undefined, args.slice(func.length).concat(cb));
 }
 
 // list devices
 function list(cb) {
-  common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/device/list", {}, "Error listing devices: ", function(err, devices){
-    if(err) return cb(err);
-    return cb(undefined, devices);
-  });
+  common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/device/list",
+    {}, "Error listing devices: ", cb);
 }
 
 function update(id, label, cb) {
-  common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/device/update", {"cuid": id, "name": label}, "Error updating device: ", function(err, data){
-    if(err) return cb(err);
-    return cb(undefined, data);
-  });
+  common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/device/update",
+    {"cuid": id, "name": label}, "Error updating device: ", cb);
 }
 
 function read(id, cb) {
-  common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/device/read", {"cuid": id}, "Error reading device: ", function(err, data){
-    if(err) return cb(err);
-    return cb(undefined, data);
-  });
+  common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/device/read",
+    {"cuid": id}, "Error reading device: ", cb);
 }
 
 function enabledDevice(id, enabled, cb) {
   var disabled = false;
-  if(enabled === "false"){
+  if(enabled === "false") {
```

Comment:
you should convert "false" to boolean false here and remove redundant disabled flag